types.cc

```diff
@@ -352,14 +352,12 @@ static uint32_t serialize(const std::string& input, int64_t days) {
     return static_cast<uint32_t>(days);
 }
 uint32_t simple_date_type_impl::from_sstring(sstring_view s) {
-    std::string str;
-    str.resize(s.size());
-    std::transform(s.begin(), s.end(), str.begin(), ::tolower);
```

Comment:
Why is this unnecessary? Changelog forgot to mention.

Reply:
Well, I see the regex below doesn't mention any letters. But it would be nice for the changelog to say so.

Reply:
Even though the regex doesn't use letters, we end up calling other functions that parse parts of it - maybe those  that need to understand day name, month names, etc., do care about capitalization? I don't know, I am not familiar with any of these functions. @nuivall please investigate what these functions are actually doing, and explain in the commit message why the tolower (which has been in this code for years) is not needed.
Even better would be to add a unit test (in C++) to test this function. Or maybe we already have one?

Reply:
added explanation to the commit message

> Even better would be to add a unit test (in C++) to test this function.

Don't think it makes sense to add any here as I don't even know where behavior of this type is specified. If there are any violations in current behavior then existing test should catch it. There are some in test/cql-pytest/test_type_*.

The only place with docs I found is https://cassandra.apache.org/doc/latest/cassandra/cql/types.html#dates but according to it even this regex is not needed as format such as "-10-40-20" is not described there.

Reply:
I finally understand. This date type is just "yyyy-mm-dd", and is only numerals. There are no month names or anything like that. Ok.

Reply:
ok, so we extend support for this from timestamp type to also date type.

Reply:
or I misinterpret `For timestamps` wording in those docs...

Reply:
It's very unclear why they referred to timestamps :-(
When this happens to me, I usually write a test to see what's really going on... For example I wrote test/cql-pytest/test_type_timestamp.py for the "timestamp" type (don't confuse this with USING TIMESTAMP!), but never did this for the "date" type.